src/__test__/client.test.ts

```diff
@@ -0,0 +1,213 @@
+      )
+      expect(info.avoid_reuse).toEqual(params.avoid_reuse || false)
+      expect(info.scanning).toEqual(false)
+    }
+  })
+  test('(un)lockwallet', async () => {
+    const walletName = 'testlock'
+    const walletPass = 'pass'
+
+    await client.createWallet(walletName, undefined, undefined, walletPass)
+    const walletClient = new Client({
+      baseURL: url + '/wallet/' + walletName,
+      auth,
+    })
+    await walletClient.walletPassphrase(walletPass, 10)
+    let info = await walletClient.getWalletInfo()
+    expect(info.unlocked_until).toBeGreaterThan(0)
+    await walletClient.walletLock()
+    info = await walletClient.getWalletInfo()
+    expect(info.unlocked_until).toEqual(0)
+  })
+  test('getnetworkinfo', async () => {
+    await expect(client.getNetworkInfo()).resolves.toBeDefined()
+  })
+  test('sendrawtransaction', async () => {
+    const walletName = 'sendraw'
+    await client.createWallet(walletName)
+    const walletClient = new Client({
+      baseURL: url + '/wallet/' + walletName,
+      auth,
+    })
+    const address = await walletClient.getNewAddress()
+    const privKey = await walletClient.dumpPrivKey(address)
+    const toAddress = await walletClient.getNewAddress()
+    await walletClient.generateToAddress(101, address)
+    const unspent = await walletClient.listUnspent()
+    const amount = unspent[0].amount * 100000000 - 110
+    const req: cfdjs.CreateRawTransactionRequest = {
+      version: 1,
+      txins: [
+        {
+          txid: unspent[0].txid,
+          vout: unspent[0].vout,
+        },
+      ],
+      txouts: [
+        {
+          address: toAddress,
+          amount,
+        },
+      ],
+    }
+
+    const rawTx = cfdjs.CreateRawTransaction(req).hex
+
+    const result = await walletClient.signRawTransactionWithKey(rawTx, [
+      privKey,
+    ])
+
+    const decoded = cfdjs.DecodeRawTransaction({ hex: result.hex })
+
+    await walletClient.sendRawTransaction(result.hex)
+
+    await walletClient.generateToAddress(1, address)
+
+    const tx = await walletClient.getTransaction(decoded.txid)
+
+    expect(tx.confirmations).toEqual(1)
+  })
+  test('using unknown wallet getwalletinfo throws', async () => {
+    const walletClient = new Client({
+      baseURL: url + '/wallet/' + 'unknown',
+      auth,
+    })
+    await expect(walletClient.getWalletInfo()).rejects.toThrowError(
+      new BitcoinRpcError(
+        RPCErrorCode.RPC_WALLET_NOT_FOUND,
+        'Requested wallet does not exist or is not loaded'
+      )
+    )
+  })
+  test('using bad credentials throws AuthError', async () => {
+    const badCredClient = new Client({
+      baseURL: url,
+      auth: { username: 'a', password: 'b' },
+    })
+    await expect(badCredClient.getNetworkInfo()).rejects.toThrow(AuthError)
+  })
+  test('non connected node throws connection error', async () => {
+    const notConnectedClient = new Client({
+      baseURL: 'http://localhost:1234',
```

Comment:
what if I'm using this port ?, maybe ensure that is different from the correct one above

Reply:
Then the test fails and you know something is wrong? At least on CI things are ok. I don't want to do something to complicated (like check which ports are open or something like that).